cmd/notification/rm/rm.go

```diff
@@ -24,32 +26,36 @@ import (
 	"github.com/spf13/cobra"
 )
 
-// NewCommand returns the rm command of type cobra.Command
-var byAge bool
+var age string
+var slug string
 
-//TODO need revisit ! What are the supported flags ? Are they consistent with the other commands ?
 func removeNotificationHandler(cmd *cobra.Command, args []string) (err error){
-	// Checking for args
-	if len(args) == 0 {
-		return errors.New("no notification ID/Name provided")
-	}
-	url := config.Conf.Clients["Notification"].Url() + clients.ApiNotificationRoute
-	if byAge {
-		url += "/age/" + args[0]
-	} else {
-		url += "/slug/" + args[0]
+	if age == "" && slug == "" {
+		return errors.New("at least one flag should be provided")
+	} else if age != "" && slug != ""{
+		return errors.New("age or slug flag should be provided, not both")
 	}
-	return request.DeletePrt(url, args[0])
+	url, deletedBy := constructUrl()
+	return request.DeletePrt(url, deletedBy)
 }
 
+// NewCommand returns the rm command of type cobra.Command
 func NewCommand() *cobra.Command {
 	var cmd = &cobra.Command{
-		Use:   "rm [notification slug or age]",
+		Use:   "rm [--slug | --age]",
 		Short: "Removes notification by slug or age",
-		Long:  `Removes a notification given its slug or age timestamp.`,
-		Args:  cobra.ExactValidArgs(1),
+		Long:  "Removes notifications by slug or age (the current notifications timestamp minus their last modification timestamp should be less than the age parameter)",
 		RunE:   removeNotificationHandler,
 	}
-	cmd.Flags().BoolVar(&byAge, "age", false, "Remove by age")
+	cmd.Flags().StringVar(&age, "age", "", "Notification age (in milliseconds)")
```

Comment:
does it make sense to change milliseconds to minutes? what is the most likely use case?
Might someone want to delete all notifications from yesterday or an hour back? translating to milli seconds may be a pain?

Reply:
I have added another optional flag called --unit that could accept time units  ms, s , m , h, d ... 
The default value of this flag is ms (milliseconds) 

Examples delete by age: 
./edgex-cli notification rm --age 1 --unit h  (for 1 hour)
./edgex-cli notification rm --age 45 --unit m  (for 45 minutes)
./edgex-cli notification rm --age 3 --unit d  (for  3 days)

Example remove notification by slug:
./edgex-cli notification rm --slug slug_name